Create WooFoundation library and move CurrencyFormatter

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->
Sorry for the big PR, most of the changes are necessary one-line import additions that cannot be broken into more than one PR.
Part of: #5108 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR is part of moving `CurrencyFormatter` to a new shared library and using it in `Yosemite`. Here we create a new module (~~Tools~~ `WooFoundation`), move `CurrencyFormatter` and related classes, and add the necessary imports for the project and tests to build and pass correctly. The tasks for the overall issue are:

1. Create `WooFoundation` library and move `CurrencyFormatter` there. (This PR)
2. Break the dependency between Tools and `Yosemite` so `WooFoundation` can be imported into `Yosemite` without a circular reference.
3. Move tests to `WooFoundation`. It needs to be done after the second point because at the moment they have a `Networking` dependency related to `Yosemite.SiteSetting` that will be eliminated in 2.
4. Use `CurrencyFormatter` in `Yosemite`

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Please test that the app builds fine, the currency formatting works properly and unit tests pass as before.

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

toupper committed:
Create Tools library and import it where appropriate

toupper committed:
Remove not needed Hardware reference

toupper committed:
Delete duplicated extensions

toupper committed:
Make tests build and pass by importing Tools

toupper committed:
Add documentation

koke commented:
> Here we create a new module (Tools)

In keeping with Apple style and tradition, can I suggest `WooFoundation` for this new library? 😁

koke reviewed (approved):
Other than the naming suggestion, this seems to build and run fine 🚀

toupper committed:
Rename new library to WooFoundation

toupper committed:
Delete old files

toupper committed:
Add WooFoundation to swiftlint

toupper commented:
Thanks for the review @koke! Indeed, `WooFoundation` sounds cooler 😄 , I changed in https://github.com/woocommerce/woocommerce-ios/pull/6973/commits/ee7978cff3985f5bef8326d9e3d3017b3def786e and added it in swiflint as well.

toupper committed:
Remove local reference

toupper committed:
Rename product bundle identifier for WooFoundationTests

toupper committed:
Remove pods for old name library

toupper committed:
Remove old name framework

toupper committed:
Add more necessary elements to WooFoundation library

toupper committed:
Avoid duplicate output file by deleting this added plist file
